### PR TITLE
fix: spring sdk remove clients secrets

### LINKED ISSUE #18656: Remove client's secrets reference in logs
Remove all the client's secretes in logs

Support case: [SUPPORT-22005](https://jira.camunda.com/browse/SUPPORT-22005)

### PR DESCRIPTION
**Pull request by nicpuppa**
## Related issues

closes #18656

### CODE REVIEW

File: spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java [hunk 1]
```diff
@@ -805,10 +805,10 @@ public String toString() {
           + clusterId
           + '\''
           + ", clientId='"
-          + clientId
+          + "***"
```
Comment (korthout) [inline spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java:808]: 👍 I like the pragmatic solution

❓ I understand we redact the `clientSecret` as it is a secret. But the `clientId` is not really secret information and may help troubleshoot configurations. Is there a specific reason we're also redacting the `clientId`?

**Review by korthout — CHANGES_REQUESTED** [review]
❌ A test should be added to show that the secret configs are not logged on startup

File: spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java (hunk 1 above)
Comment (nicpuppa) [reply spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java:808]: We are doing [that](https://github.com/camunda/zeebe/blob/0d2e6d277efb00b4f3ca932764f025470c578259/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/common/Client.java#L35-L36) for SM clusters, so I thought to have the same behaviour also for Saas

Comment (korthout) [reply spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java:808]: Ah, I see. Thanks @nicpuppa 👍 I guess it doesn't hurt too much if we have clear config properties and documentation. Makes sense to do it consistently.

File: spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplOidcTest.java [hunk 2]
```diff
@@ -169,4 +170,10 @@ void shouldHaveDefaultRetryPolicy() {
     assertThat(zeebeClientConfiguration.useDefaultRetryPolicy())
         .isEqualTo(DEFAULT.useDefaultRetryPolicy());
   }
+
+  @Test
+  void shouldNotLogClientInfoAtStartup(final CapturedOutput output) {
+    assertThat(output).contains("clientId='***'");
+    assertThat(output).contains("clientSecret='***'");
+  }
```
Comment (korthout) [inline spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplOidcTest.java:178]: 😍

File: spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplOidcTest.java [hunk 3]
```diff
@@ -27,6 +27,7 @@
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.system.CapturedOutput;
 
 @SpringBootTest(
     classes = {ZeebeClientAllAutoConfiguration.class, ZeebeClientProdAutoConfiguration.class},
```
Comment (korthout) [inline spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplOidcTest.java:33]: 🤔 Does this class need this annotation like the other test class?

>`@ExtendWith(OutputCaptureExtension.class)`

**Review by korthout — APPROVED** [review]
Thanks @nicpuppa 👍

File: spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplOidcTest.java (hunk 3 above)
Comment (nicpuppa) [reply spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplOidcTest.java]: You were faster then me 👍

**Pull request merged**